vital/types/object_track_set.h

```diff
@@ -104,20 +106,7 @@ class VITAL_EXPORT object_track_state : public track_state
   object_track_state( object_track_state&& other ) = default;
 
   /// Clone the track state (polymorphic copy constructor)
-  track_state_sptr clone( clone_type ct = clone_type::DEEP ) const override
```

Comment:
Some additional comments could help here. What are the option settings and how does that effect the result?

Reply:
The options are just deep and shallow, which are pretty standard.

I did move the implementation from the .h to the .cxx, maybe its better to keep in the header?

Reply:
See also #812. Note that the default needs to be `DEEP`, otherwise it is hard to generically and *safely* clone a KWIVER object. As noted, "deep" and "shallow" are generally understood, but I guess if documentation is wanted, it should be on the `enum`. To wit, a shallow clone can potentially be mutated by mutating the original.

I think moving to the `.cxx` is fine, especially as I think it often can't be inlined anyway (often it will get called via virtual dispatch). I think I only had it in the header because it was there prior to #812. (@mleotta might have an opinion?)

Reply:
Generally I prefer the implementation in the .cxx file unless there is some mitigating factors like it is a template.